pyuvdata/utils.py

```diff
@@ -1038,7 +1040,1316 @@ def unphase_uvw(ra, dec, uvw):
+            new_coords = rotate_two_axis(
+                rotate_two_axis(  # Yo dawg, I heard you like rotation maticies...
+                    uvw_array[:, [2, 0, 1], np.newaxis],
+                    0.0 if from_enu else (-old_app_pa),
+                    (-telescope_lat) if from_enu else (-old_app_dec),
+                    0,
+                    1,
+                ),
+                gha_delta_array,
+                telescope_lat if to_enu else app_dec,
+                2,
+                1,
+            )
+
+            # One final rotation applied here, to compensate for the fact that we want
+            # the Dec-axis of our image (Fourier dual to the v-axis) to be aligned with
+            # the chosen frame
+            new_coords = rotate_one_axis(new_coords, app_pa, 0)[:, :, 0]
+
+    # There's one last task to do, which is to re-align the axes from projected
+    # XYZ -> uvw, where X (which points towards the source) falls on the w axis,
+    # and Y and Z fall on the u and v axes, respectively.
+    return new_coords[:, [1, 2, 0]]
+
+
+def translate_driftscan_to_app(az_val, el_val, telescope_lat, lst_array):
+    """
+    Translate a fixed az-el pointing into apparent RA/Dec coordinates.
+
+    Calculates apparent coordinates for a driftscan observation, helpful (?) for
+    telescopes which do not track.
+
+    Parameters
+    ----------
+    az_val : float
+        Azimuth value of the pointing, in radians.
+    el_val : float
+        Elevation value of the pointing, in radians.
+    telescope_lat : float
+        Latitude of the observatory, in radians.
+    lst_array : ndarray of float
+        LSTs at the time of observations, in units of radians.
+    """
+    coord_vector = np.array(
+        [
+            [
+                np.sin(el_val),  # X points up
+                -np.sin(az_val)
+                * np.cos(el_val),  # Flip Y due to l-hand -> r-hand in az->ha
+                np.cos(az_val) * np.cos(el_val),  # Z is north-south
+            ]
+        ]
+    )
+    # Rotate the vector based on the telescope latitude
+    new_vector = rotate_one_axis(coord_vector, (np.pi / 2.0) - telescope_lat, 1)
+
+    # RA vector is a little more complicated, though not by much. The new vector
+    # gives us the hour angle, and so we just have to add the lst to that
+    app_ra = np.arctan2(new_vector[2], new_vector[1]) + lst_array
+
+    # Dec vector is easy -- it remains fixed for all time
+    app_dec = np.zeros_like(lst_array) + np.arcsin(new_vector[2])
+
+    return app_ra, app_dec
+
+
+def translate_ephem_to_app():
+    """
+    Translate a TLE ephem (e.g., for a planet) into apparent of RA and Dec.
+
+    This is a thing that does not yet exist.
+    """
+    # TODO: Karto needs to introduce some optional dependency pieces here.
+    # Don't let him slack off!
+    raise NotImplementedError("Support for solar-system objects not available (yet).")
+
+    return None, None
+
+
+def translate_sidereal_to_sidereal(
```

Comment:
I've renamed it `translate_btw_sidereal_frames` -- hopefully that's a bit more descriptive.

Reply:
This is a bit of a nit pick, but "translate" has a specific meaning of changing position withOUT rotation, which is clearly not what this function does.  Perhaps `transform_sidereal_frames` would be more apt.  BTW, I'm not sure how I feel about "btw" in the name.  I would vote to either spell it out or leave to out altogether (favoring the latter since the plural "frames" and the to/from arguments do a reasonable job of implying that IMHO).  I guess one could even drop the plural since it's only from a single frame to a single frame (I think).